### PR TITLE
Handle Errors Gracefully

### PR DESCRIPTION
**Pull request by Duued**
Focusing on handling problems gracefully, attempting to work around, and telling the user what went wrong (along with steps to resolve.)  If there is no possible way to work around it, the program will stop running and tell the user what went wrong. (Such as missing label, no WRAP link, etc.)

Note: I don't know why it said I deleted nearly all of the contents of spotify_wrapped.py... Sorry about that.

### CODE REVIEW

File: spotify_wrapped.py [hunk 1]
```diff
@@ -1,128 +1,172 @@
...
+
+pandas_url = convert_google_sheet_url(google_sheets_link)
+
+df = pd.read_csv(pandas_url)
+artist = CONFIG['columnC-name']
+try:
+    counts = Counter(df[artist])
+    wrapped_artist = CONFIG['columnC-name']
+    wrapped_song = CONFIG['columnB-name']
+except (KeyError, AttributeError):
```
Comment (Duued) [inline spotify_wrapped.py:45]: Two errors can occur. Catch them to tell user what went wrong.

File: spotify_wrapped.py [hunk 2]
```diff
@@ -1,128 +1,172 @@
...
+    print("Please check your google spreadsheet and ensure the headers both exist and match the config.")
+    quit()
+
+
+df_date = df[CONFIG['columnA-name']]
```
Comment (Duued) [inline spotify_wrapped.py:50]: Change `df.date` to `df_date`

File: spotify_wrapped.py [hunk 3]
```diff
@@ -1,128 +1,172 @@
...
+    print(f"JULY SONG NUMBER: {len(df[df_date.str.contains('July')])} (ROUGHLY {3*len(df[df_date.str.contains('July')]) / 60} HOURS)")
+    print(f"AUGUST SONG NUMBER: {len(df[df_date.str.contains('August')])} (ROUGHLY {3*len(df[df_date.str.contains('August')]) / 60} HOURS)")
+    print(f"SEPTEMBER SONG NUMBER: {len(df[df_date.str.contains('September')])} (ROUGHLY {3*len(df[df_date.str.contains('September')]) / 60} HOURS)")
+    print(f"OCTOBER SONG NUMBER: {len(df[df_date.str.contains('October')])} (ROUGHLY {3*len(df[df_date.str.contains('October')]) / 60} HOURS)")
+    print(f"NOVEMBER SONG NUMBER: {len(df[df_date.str.contains('November')])} (ROUGHLY {3*len(df[df_date.str.contains('November')]) / 60} HOURS)")
+    print(f"DECEMBER SONG NUMBER: {len(df[df_date.str.contains('December')])} (ROUGHLY {3*len(df[df_date.str.contains('December')]) / 60} HOURS)")
+
+print("\n")
+
+counts_1 = Counter(wrapped[wrapped_artist])
```
Comment (Duued) [inline spotify_wrapped.py:70]: Change `counts_1` and `counts_2` to respect CONFIG

File: spotify_wrapped.py [hunk 4]
```diff
@@ -1,128 +1,172 @@
...
+values_list_artist = list(most_popular_artist.values())
+
+#print(f"ARTISTS WITH MORE THAN 10 PLAYS IN {current_month}:\n")
+
+print("MY TOP TEN ARTISTS ON SPOTIFY OF 2024")
+
+try:
+    for i in range(0, 10): #range(len(keys_list_artist)): # Provides your top ten artists, if you want all artists more >= 10, change range to commented
+        print(values_list_artist[i], keys_list_artist[i])
+except IndexError:
```
Comment (Duued) [inline spotify_wrapped.py:106]: If you provide a number out of bounds, this error will occur. Catch it to tell the user what went wrong.

**Comment by liz-stippell** [comment]
PRs will be accepted if they add new files to the repo without changing the original `spotify_wrapped.py`. Please use some other naming scheme for your edits to the script (e.g., `spotify_wrapped_[GH-name].py`, `spotify_wrapped_handle_errors.py`, etc.). This precaution ensures there are no bugs and others can still use the original script. Please also provide a detailed description of what you changed/improved in the code when you commit changes (this description will be copied and pasted into the ReadMe file so that others know what it does; line numbers of changes/additions and what they are is encouraged but not required).

**Pull request closed without merging**